crates/fast_compute/src/layer.rs

```diff
@@ -539,6 +540,23 @@ impl<'a, T: TowerFamily, P: PackedTop<T>> ComputeLayerExecutor<T::B128>
 
 		Ok((out1?, out2?))
 	}
+
+	fn map<Out: Send, I: ExactSizeIterator<Item: Send> + Send>(
+		&mut self,
+		iter: I,
+		map: impl Sync + Fn(&mut Self, I::Item) -> Result<Out, Error>,
+	) -> Result<Vec<Out>, Error> {
+		// `par_bridge` doesn't preserve the order of the items in the iterator,
+		// so we need to enumerate the items and sort them back after processing.
+		let mut result = iter
+			.enumerate()
+			.par_bridge()
+			.map(|(index, item)| (index, map(&mut self.clone(), item)))
```

Comment:
Would it be possible to pass an implementation-defined executor instance to this callback? Doing so would avoid requiring `Sync` on `ComputeLayer`, which is not required by the ic1 implementation.